### ISSUE TITLE
[Feature Request] Hard Line Break Support

### ISSUE THREAD

**Issue by KSR-Yasuda** [previm/previm#163]
Couldn't you add hard line break option?

Currently, Previm follows the original Markdown syntax in line break,  
i.e. *space-space-LF* is considered as line break (soft line break).

Commonly used 'GitHub Flavored Markdown' considers  
just putting *LF* as a line break (hard line break).

So I wonder if there's an option to switch soft / hard line break.

----

I know that changing like below changes to hard line break mode.  
Though, it is a bit bothering to switch it easily.

Could Previm pass some setting value there somehow?

```diff
--- a/preview/_/js/previm.js
+++ b/preview/_/js/previm.js
@@ -2,7 +2,7 @@
 
 (function(_doc, _win) {
   var REFRESH_INTERVAL = 1000;
-  var md = new _win.markdownit({html: true, linkify: true})
+  var md = new _win.markdownit({html: true, linkify: true, breaks: true})
                    .use(_win.markdownitAbbr)
                    .use(_win.markdownitDeflist)
                    .use(_win.markdownitFootnote)
```

**Comment by KSR-Yasuda** [comment]
Patch below passes `g:previm_hard_line_break` value on every preview rendering,  
though, LF between CJK characters isn't considered as a line break
(e.g. `ほげ\nふが` is rendered as `ほげふが`, without line breaking).

Perhaps because of `markdown-it-cjk-breaks` module?

```diff
diff --git a/autoload/previm.vim b/autoload/previm.vim
index 2ed94a8..4379156 100644
--- a/autoload/previm.vim
+++ b/autoload/previm.vim
@@ -137,6 +137,10 @@ function! s:function_template() abort
       \ printf('return %s;', get(g:, 'previm_show_header', 1)),
       \ '}',
       \ '',
+      \ 'function isHardLineBreak() {',
+      \ printf('return %s;', (get(g:, 'previm_hard_line_break', v:false) ? 'true' : 'false')),
+      \ '}',
+      \ '',
       \ 'function getFileName() {',
       \ printf('return "%s";', s:escape_backslash(current_file)),
       \ '}',
```

```diff
diff --git a/preview/_/js/previm.js b/preview/_/js/previm.js
index 84f1ae7..29173cd 100644
--- a/preview/_/js/previm.js
+++ b/preview/_/js/previm.js
@@ -24,6 +24,9 @@
 
   function transform(filetype, content) {
     if(hasTargetFileType(filetype, ['markdown', 'mkd'])) {
+      if (typeof isHardLineBreak === 'function') {
+        md.set({ breaks: isHardLineBreak() });
+      }
       return md.render(content);
     } else if(hasTargetFileType(filetype, ['rst'])) {
       // It has already been converted by rst2html.py
```

**Comment by KSR-Yasuda** [comment]
Disabling `markdown-it-cjk-breaks` on hard line break mode looks working.

```diff
--- a/preview/_/js/previm.js
+++ b/preview/_/js/previm.js
@@ -24,8 +24,12 @@
 
   function transform(filetype, content) {
     if(hasTargetFileType(filetype, ['markdown', 'mkd'])) {
-      if (typeof isHardLineBreak === 'function') {
-        md.set({ breaks: isHardLineBreak() });
+      if (typeof isHardLineBreak === 'function' && isHardLineBreak()) {
+        md.set({ breaks: true });
+        md.disable('cjk_breaks');
+      } else {
+        md.set({ breaks: false });
+        md.enable('cjk_breaks');
       }
       return md.render(content);
     } else if(hasTargetFileType(filetype, ['rst'])) {
```

**Comment by mattn** [comment]
Closed by #164
